s64da_benchmark_toolkit/prepare.py

```diff
@@ -38,6 +43,30 @@ def __init__(self, args, benchmark):
         self.data_dir = args.data_dir
         assert os.path.isdir(self.schema_dir), 'Schema does not exist'
 
+    @property
+    def swarm64da_version(self):
+        try:
+            with DBConn(self.args.dsn) as conn:
+                conn.cursor.execute('SELECT swarm64da.get_version()')
+                result = conn.cursor.fetchone()[0]
+
+            s64da_version_string = re.findall(r'[0-9]+\.[0-9]+\.[0-9]+', result)[0]
+            return Swarm64DAVersion(*[int(part) for part in s64da_version_string.split('.')])
+
+        except ProgrammingError:
+            return None
+
+    @property
+    def supports_cluster(self):
+        version = self.swarm64da_version
+        if not version:
+            return False
+
+        if version.major <= 4 and version.minor < 1:
```

Comment:
This would make 3.1 support `cluster`. You probably want a lexicographical compare, something like `if version.major < 4 or version.major == 4 and version.minor < 1:`

Reply:
Yes ... exactly what I was looking for. Thanks a bunch.

Reply:
I know this has already been merged, but have you considered using https://packaging.pypa.io/en/latest/version/#packaging.version.parse ?

Reply:
Yes! Good one! Will add it to the next one.